src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -0,0 +1,338 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.Interop
+{
+    [Generator]
+    public sealed class ComInterfaceGenerator : IIncrementalGenerator
+    {
+        public static class StepNames
+        {
+            public const string CalculateStubInformation = nameof(CalculateStubInformation);
+            public const string GenerateManagedToNativeStub = nameof(GenerateManagedToNativeStub);
+            public const string GenerateNativeToManagedStub = nameof(GenerateNativeToManagedStub);
+        }
+
+        private static readonly ContainingSyntax NativeTypeContainingSyntax = new(
+                                    TokenList(Token(SyntaxKind.InternalKeyword), Token(SyntaxKind.PartialKeyword)),
+                                    SyntaxKind.InterfaceDeclaration,
+                                    Identifier("Native"),
+                                    null);
+
+        public void Initialize(IncrementalGeneratorInitializationContext context)
+        {
+            // Get all methods with the [GeneratedComInterface] attribute.
+            var attributedMethods = context.SyntaxProvider
+                .ForAttributeWithMetadataName(
+                    TypeNames.GeneratedComInterfaceAttribute,
+                    static (node, ct) => node is InterfaceDeclarationSyntax,
+                    static (context, ct) => context.TargetSymbol is INamedTypeSymbol methodSymbol
+                        ? new { Syntax = (InterfaceDeclarationSyntax)context.TargetNode, Symbol = methodSymbol }
```

Comment:
Is having the symbol in the data is going to keep the previous compilations alive?

Reply:
Not in this case. We aren't using a custom comparer, so each time we get a new compilation, we'll replace the symbol object here with a symbol from the new compilation.

Reply:
In that case, won't the steps that have Symbols as part of the output never be incremental since the symbols won't compare as equal? Not that we need to fix this right away, just want to make sure I understand this.

Reply:
Yes, that is true. Steps that output symbols are not incremental. The generators aren't incremental until the step where we build the context that doesn't have symbols. It's not perfect, but we still have incremental behavior from that point onwards, so it's "fine".